Don't throw exception if AD user does not exist

maxxer opened the pull request:
This patch (which can be squashed) allows the creation of Zimbra mailboxes without corresponding user in AD. Specifically allows setting user's password into Zimbra's LDAP, without throwing an LDAP error for the missing AD user.

Commit:
don't throw exception if user does not exist

maxxer committed:
Fix user existance check

ghost commented:
Put also if anything else goes wrong aka the search fails because of config issues or such, no error will be logged....

maxxer committed:
Honor zimbraAuthFallbackToLocal when checking AD user

If it's FALSE then throws an exception when changing pw for user not
present in AD. Otherwise just logs a warning.

maxxer commented:
I've added a different behaviour based on `zimbraAuthFallbackToLocal` value. If FALSE throws exception. Also added explaination in README.

What do you think?

maxxer committed:
Minor changes to README about debugging

maxxer commented:
I did also a small change in README about debugging: I didn't find messages in `mailbox.log` but in `zmmailboxd.out`. Is it correct?

ghost commented:
it does log to zmmailboxd.out yes. Merged.
